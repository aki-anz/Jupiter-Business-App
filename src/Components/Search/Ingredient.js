import React from 'react';
import {v4 as uuidv4} from 'uuid';
import styled from '@emotion/styled';

const List = styled.ul`
    margin-top: 1rem;
    text-decoration: none;
    list-style-type: none;
    border-radius: .6rem;
    background-color: #90EE90;
    padding: .3rem;
    margin: .3rem;
    color: #663300;
`

const Text = styled.li`
    font-size: 1rem;
`

const Ingredient = ({ingredients}) => {
    return ingredients.map(ingredient => {
    return (
        <List key={uuidv4()}>
            <Text className='ingredient-text'>{ingredient.text}</Text>
        </List>
    );
    });
}

export default Ingredient;
