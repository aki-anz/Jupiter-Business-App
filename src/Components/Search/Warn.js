import React from 'react';
import styled from '@emotion/styled';

const Message = styled.div`
    position: absolute;
    top: 6rem;
    left: 2.6rem;
    width: calc(100% - 5rem);
    border-radius: .4rem;
    padding: .2rem;
`

const ErrorText = styled.p`
    font-size: 1.4rem;
    font-weight: bolder;
    color: red;
    text-transform: uppercase;
    text-align: center;
`

const Warn = ({warn}) => {
    return (
        <Message>
            <ErrorText>{warn}</ErrorText>
        </Message>
    );
}

export default Warn;
