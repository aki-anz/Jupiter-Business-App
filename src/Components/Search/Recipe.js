import React, {useState} from 'react';
import Ingredient from './Ingredient';
import styled from '@emotion/styled';

const Card = styled.div`
    display: flex;
    flex-direction: column;
    position: relative;
    width: 20rem;
    margin: 2rem;
    padding: 1rem;
    border-radius: .6rem;
    background-color: #90EE90;
`

const Title = styled.h3`
    position: absolute;
    padding: .5rem;
    top: 1rem;
    left: 1.5rem;
    right: .5rem;
    font-size: 1rem;
    color: #1d1d1d;
`

const Image = styled.img`
    border-radius: .4rem;
    object-fit: cover;
`

const Link = styled.a`
    font-size: 1rem;
    color: #2a5be2;
    border: none;
    padding: .2rem;
    &:hover {
        cursor: pointer;
    }
    &:focus {
        outline: none;
    }
`

const Show = styled.button`
    font-size: 1rem;
    text-transform: uppercase;
    border: none;
    border-radius: .4rem;
    height: 2rem;
    background-color: coral;
    color: white;
    &:hover {
        cursor: pointer;
        background-color: white;
        color: coral;
        transition: ease-in-out .2s;
    }
`

const Recipe = ({recipe}) => {
    const [show, setShow] = useState(false);
    const {label, image, url, ingredients} = recipe.recipe;

    const onClick = () => {
        setShow(!show);
    }

    return (
        <Card>
            <Title>{label}</Title>
            <Image src={image} alt={label} />
            <Link href={url} target='_blank' rel='noopener noreferrer'>link</Link>
            <Show onClick={onClick}>Ingredients</Show>
            {show && <Ingredient ingredients={ingredients} />}
        </Card>
    );
}

export default Recipe;
