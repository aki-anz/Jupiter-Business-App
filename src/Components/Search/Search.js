import React, {useState} from 'react';
import './Search.css';
import Axios from 'axios';
import {v4 as uuidv4} from 'uuid';
import Recipe from './Recipe';
import Warn from './Warn';
import styled from '@emotion/styled';

const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`
const Form = styled.form`
    position: relative;
    width: 30rem;
    height: 8rem;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 0.5rem;
    margin-bottom: .5rem;
`
const Field = styled.input`
    font-size: .8rem;
    border: none;
    margin: .8rem;
    border-radius: .4rem;
    height: 2.5rem;
    padding: 0.5rem;
    width: 35%;
    &:focus {
        outline: none;
    }
`
const Submit = styled.input`
    width: 10%;
    height: 2rem;
    text-transform: uppercase;
    align-items: center;
    border: none;
    font-size: 1rem;
    border-radius: .4rem;
    background-color: #2a5be2;
    color: white;
    &:hover {
        cursor: pointer;
    }
`
const Recipes = styled.div`
    width: 80%;
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    justify-content: center;
    align-items: flex-start;
`

const Search = () => {
    const [query, setQuery] = useState('');
    const [recipes, setRecipes] = useState([]);
    const [warn, setWarn] = useState('');
   
    const APP_ID = 'a0e4f85b';
    const APP_KEYS = '63ef50148e6cd39b105692645017dc95';
    const url = `https://api.edamam.com/search?q=${query}&app_id=${APP_ID}&app_key=${APP_KEYS}`;

    const getData = async() => {
        if (query !== '') {
        const result = await Axios.get(url);
        setRecipes(result.data.hits);
        setQuery('');
        setWarn('');
    } else {
        setWarn('Search field is empty!');
    }
    }

    const onChange = (e) => {
        setQuery(e.target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        getData();
    }

    return (
        <Container>
            <Form onSubmit={onSubmit}>
                {warn !== '' && <Warn warn={warn} />}
                <Field
                type='text'
                name='query'
                onChange={onChange}
                value={query}
                placeholder='Enter Search Item...'
                data-testid='field'
                />
                <Submit type='submit' value='Go' data-testid='btn' />
            </Form>
            <Recipes>
            {
                recipes !== [] && recipes.map(recipe => <Recipe key={uuidv4()} recipe={recipe} />)
            }
            </Recipes>
        </Container>
    );
}

export default Search;
