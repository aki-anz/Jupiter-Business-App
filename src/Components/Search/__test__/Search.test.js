import React from 'react';
import ReactDOM from 'react-dom';
import {render, fireEvent, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Search from '../Search';

afterEach(cleanup);

it('Component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search />, div);
    ReactDOM.unmountComponentAtNode(div);
})

it('Form renders properly', () => {
    const {getByTestId} = render(<Search />);
    const field = getByTestId('field');
    const btn = getByTestId('btn');
    expect(field).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
})

it('input updates on change', () => {
    const {getByTestId} = render(<Search />);
    const inputField = getByTestId('field');
    fireEvent.change(inputField, {target: {value: 'test item'}});
    expect(inputField.value).toBe('test item');
})

it('Component matches snapshot', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
})

