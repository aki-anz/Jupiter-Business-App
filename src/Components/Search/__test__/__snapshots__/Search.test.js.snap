// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<div
  className="css-tjuh7q"
>
  <form
    className="css-iqyrq2"
    onSubmit={[Function]}
  >
    <input
      className="css-o077wi"
      data-testid="field"
      name="query"
      onChange={[Function]}
      placeholder="Enter Search Item..."
      type="text"
      value=""
    />
    <input
      className="css-1b4d6h5"
      data-testid="btn"
      type="submit"
      value="Go"
    />
  </form>
  <div
    className="css-wmmdcp"
  />
</div>
`;
